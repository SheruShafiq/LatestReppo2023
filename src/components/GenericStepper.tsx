import {
  Step,
  StepButton,
  StepLabel,
  Stepper,
  Typography,
} from "@mui/material";
import React from "react";
import useResizeHandler from "../lib/hooks/useResizeHandler";

export type Step = {
  label: string;
  navigation: (index: number) => void;
  isOptional?: boolean;
};

export type GenericStepperProps = {
  activeStep: number;
  steps: Step[];
};

const GenericStepper = ({ activeStep, steps }: GenericStepperProps) => {
  const size = useResizeHandler();
  return (
    <Stepper activeStep={activeStep}>
      {steps.map((step, index) => {
        const stepProps: { completed?: boolean } = {};
        const labelProps: { optional?: React.ReactNode } = {};
        if (step.isOptional) {
          labelProps.optional = (
            <Typography variant="caption">Optional</Typography>
          );
        }
        return (
          <Step key={step.label} {...stepProps}>
            <StepButton {...labelProps} onClick={() => step.navigation(index)}>
              {size.width > 600 ? step.label : null}
            </StepButton>
          </Step>
        );
      })}
    </Stepper>
  );
};

export default GenericStepper;
