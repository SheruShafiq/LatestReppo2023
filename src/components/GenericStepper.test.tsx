import { render } from '@testing-library/react';
import { test } from 'vitest';
import useResizeHandler from "../lib/hooks/useResizeHandler";
import GenericStepper, { Step } from '../components/GenericStepper';
import { debug } from 'vitest-preview';
import matchers from "@testing-library/jest-dom";
expect.extend(matchers);

const stepsMock: Step[] = [
  { label: 'Step 1', navigation: vi.fn(), isOptional: true },
  { label:  'Step 2', navigation: vi.fn() },
];

test('renders correctly when width > 600', () => {
    global.innerWidth = 800;
    global.dispatchEvent(new Event('resize'));

  const { getByText } = render(<GenericStepper activeStep={0} steps={stepsMock} />);
  expect(getByText('Step 1')).toBeInTheDocument();
  expect(getByText('Step 2')).toBeInTheDocument();
});

test('does not render step labels when width <= 600', () => {
        // Change the viewport to 500px.
        global.innerWidth = 500;

        // Trigger the window resize event.
        global.dispatchEvent(new Event('resize'));
    
  const { queryByText } = render(<GenericStepper activeStep={0} steps={stepsMock} />);
  debug();

  expect(queryByText('Step 1')).toBeNull();
  expect(queryByText('Step 2')).toBeNull();
  expect(screen).toMatchSnapshot();
});