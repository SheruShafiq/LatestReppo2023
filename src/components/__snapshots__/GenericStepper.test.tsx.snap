// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`does not render step labels when width <= 600 1`] = `Screen {}`;
